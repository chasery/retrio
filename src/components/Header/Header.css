.Header {
  position: fixed;
  z-index: 10;
  width: 100%;
  background-color: #ffffff;
  box-shadow: 0 0.125rem 0.125rem #0000001c;
}

.Header.fullWidth .Header__wrapper {
  max-width: 100%;
}

.Header__wrapper {
  display: flex;
  align-items: center;
  padding: 1.5rem 1rem;
}

.Header__title {
  flex-shrink: 0;
  overflow: hidden;
}
.Header__title > a {
  display: flex;
  align-items: center;
  font-weight: 400;
  color: #474747;
  letter-spacing: 0.25rem;
  text-transform: uppercase;
  text-decoration: none;
}
.Header__title > a:hover {
  text-decoration: underline;
}
.Header__title > a > img {
  margin-right: 1rem;
}

.Header__subTitle {
  font-weight: 400;
  flex-shrink: 1;
  margin: 0 1rem;
  min-width: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.Header__button {
  display: block;
  padding: 0.5rem 1rem;
  color: #2660a4;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 0.25rem;
  text-decoration: none;
  transition: background-color 0.25s ease;
}
.Header__button:hover {
  box-shadow: inset 0 0.0625rem 0.0625rem #0000001c;
}
.Header__button.primary {
  background-color: transparent;
}
.Header__button.primary:hover {
  background-color: #2660a4;
  color: #fff;
}

.Nav {
  flex-grow: 1;
  flex-shrink: 0;
}
.Nav > ul {
  display: flex;
  justify-content: flex-end;
}
.Nav > ul > li + li {
  margin-left: 0.5rem;
}

@media all and (max-width: 32rem) {
  .Header__title > a {
    letter-spacing: 0.125rem;
  }
  .Header__title > a > img {
    margin-right: 0.5rem;
  }
  .Header__titleAuthed span,
  .Header__subTitle {
    display: none;
  }
  .Header__button {
    padding: 0.5rem;
  }
}

@media all and (min-width: 50rem) {
  .Header__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
