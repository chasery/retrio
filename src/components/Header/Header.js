import React from 'react';
import { Link, useLocation, useParams } from 'react-router-dom';
import './Header.css';

function Header(props) {
  const location = useLocation();

  return (
    <header className='Header' role='banner'>
      <div className='Header__wrapper'>
        <h1 className='Header__title'>
          <Link to='/'>Retrio</Link>
        </h1>
        <nav className='Nav' role='navigation'>
          {location.pathname === '/' ? (
            <ul>
              <li>
                <Link to='/'>Sign Up</Link>
              </li>
              <li>
                <Link to='/boards'>Sign In</Link>
              </li>
            </ul>
          ) : (
            <ul>
              <li>
                <Link to='/boards'>Boards</Link>
              </li>
              <li>
                <Link to='/teams'>Teams</Link>
              </li>
              <li>
                <Link to='/'>Sign Out</Link>
              </li>
            </ul>
          )}
        </nav>
      </div>
    </header>
  );
}

export default Header;
