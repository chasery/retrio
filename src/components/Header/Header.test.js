import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/',
  }),
  useParams: () => ({
    rackId: 1,
  }),
}));

describe('Header component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
