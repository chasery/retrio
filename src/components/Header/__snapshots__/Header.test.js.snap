// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without error 1`] = `
<header
  className="Header "
  role="banner"
>
  <div
    className="Header__wrapper"
  >
    <h1
      className="Header__title"
    >
      <Link
        to="/"
      >
        <img
          alt="Retrio logo"
          src="retrio-logo.svg"
        />
        <span>
          Retrio
        </span>
      </Link>
    </h1>
    <h3
      className="Header__subTitle"
    />
    <nav
      className="Nav"
      role="navigation"
    >
      <ul>
        <li>
          <Link
            className="Header__button primary"
            to="/sign-up"
          >
            Sign Up
          </Link>
        </li>
        <li>
          <Link
            className="Header__button primary"
            to="/sign-in"
          >
            Sign In
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
